generalized_accounts/generalized_accounts.md

```diff
@@ -1,9 +1,339 @@
+  2. the `nonce` included in the transaction is bigger than the nonce stored
+     in the account.
+
+
+## Transactions
+
+Two new transactions are introduced. The `ga_attach_tx` is used to convert a POA
+into a GA by attaching code to it.
+Afterwards the new `meta_tx` is used to interact with the GA.
+
+
+### `ga_attach_tx`
+
+ ```
+ Fieldname       Size (bytes)
+ -------------- -----
+| owner_id     | 32  |
+ -------------- -----
+| nonce        | 32  |
+ -------------- -----
+| code         | var |
+ -------------- -----
+| auth_fun     | var |
+ -------------- -----
+| ct_version   | 32  |
+ -------------- -----
+| fee          | 32  |
+ -------------- -----
+| ttl          | 32  |
+ -------------- -----
+| gas          | 32  |
+ -------------- -----
+| gas_price    | 32  |
+ -------------- -----
+| call_data    | var |
+ -------------- -----
+```
+
+For the actual wire format please consult the [serialisation](../serializations.md) document.
+
+The semantics of this transaction are almost identical to the [Create Contract Transaction](../contracts/contract_transactions.md#create-contract-transaction).
+As such, only the additional logic not covered by the `Create Contract
+Transaction` will be listed here.
+
+The address for the contract, which holds `code`, will be computed just like
+regular contract creation.
+
+The `owner_id` is the POA, which will be converted into a GA, if the transaction
+is valid. If the account at `owner_id` is already a GA, the transaction MUST be
+considered invalid.
+
+The transaction MUST include a signature created by the private key belonging to
+the `ownder_id`.
+
+The `auth_fun` is a function hash as described in the [Sophia](../contracts/sophia.md)
+section.
+
+If valid, the transaction will set the `ga_contract` and `ga_auth_fun` of the
+`owner_id` account.
+
+
+### `meta_tx`
+
+ ```
+ Fieldname       Size (bytes)
+ -------------- -----
+| ga_id        | 32  |
+ -------------- -----
+| auth_data    | var |
+ -------------- -----
+| abi_version  | 32  |
+ -------------- -----
+| fee          | 32  |
+ -------------- -----
+| gas          | 32  |
+ -------------- -----
+| gas_price    | 32  |
+ -------------- -----
+| ttl          | 32  |
+ -------------- -----
+| tx           | var |
+ -------------- -----
+```
+
+For the actual wire format please consult the [serialisation](../serializations.md) document.
+
+The semantics of this transaction are similar to the [Call Contract Transaction](../contracts/contract_transactions.md#contract-call-transaction).
+
+A `meta_tx` is always a call of the `auth_fun` on the contract located at
+`ga_id` with the supplied `auth_data`.
+
+The `auth_fun` call MUST return true for the `tx` to be considered valid.
+
+The `auth_fun` is a function hash as described in the [Sophia](../contracts/sophia.md)
+section.
+
+The `tx`, or inner transaction, MUST be a well formed transaction with an empty
+signature. The inner transaction MUST NOT be a `ga_attach_tx` transaction.
+
```

Comment:
The inner transaction should have nonce set to 0.

Reply:
Should or must?

Reply:
I included it in the `MUST`